src/NServiceBus.Transport.RabbitMQ/Sending/BasicPropertiesExtensions.cs

```diff
@@ -23,7 +23,15 @@ public static void Fill(this IBasicProperties properties, OutgoingMessage messag
 
             var delayed = CalculateDelay(deliveryConstraints, out var delay);
 
-            properties.Persistent = !messageHeaders.ContainsKey(UseNonPersistentDeliveryHeader);
+            if (messageHeaders.ContainsKey(UseNonPersistentDeliveryHeader))
```

Comment:
This could be replaced with `properties.Persistent = !messageHeaders.Remove(UseNonPersistentDeliveryHeader);`

Reply:
👍